error in windows 11, second

PS C:\Users\user\Documents\personal\.obsidian\plugins\SpeechNote> npm run build

> obsidian-speech-to-text@1.0.0 build
> tsc -noEmit -skipLibCheck && node esbuild.config.mjs production

src/application/EventManager.ts:311:13 - error TS2322: Type 'keyof AppEventMap' is not assignable to type 'number'.
  Type 'string' is not assignable to type 'number'.

311             totalListeners: eventNames.reduce((sum, event) =>
                ~~~~~~~~~~~~~~

  src/application/EventManager.ts:350:5
    350     totalListeners: number;
            ~~~~~~~~~~~~~~
    The expected type comes from property 'totalListeners' which is declared here on type 'EventManagerStatus'

src/application/EventManager.ts:312:17 - error TS2365: Operator '+' cannot be applied to types 'keyof AppEventMap' and 'number'.

312                 sum + this.listenerCount(event as keyof AppEventMap), 0
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/infrastructure/api/SettingsAPI.ts:285:28 - error TS2304: Cannot find name 'validation'.

285                 warnings: (validation as any).warnings?.map((w: any) => w.message)
                               ~~~~~~~~~~

src/infrastructure/api/SettingsAPI.ts:318:5 - error TS2416: Property 'on' in type 'SettingsAPI' is not assignable to the same property in base type 'EventEmitter<DefaultEventMap>'.
  Type '(event: string, listener: Function) => Unsubscribe' is not assignable to type '<K>(eventName: string | symbol, listener: (...args: any[]) => void) => this'.
    Type 'Unsubscribe' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'Unsubscribe'.

318     on(event: string, listener: Function): Unsubscribe {
        ~~

src/ui/dashboard/StatisticsDashboard.ts:192:36 - error TS2554: Expected 2 arguments, but got 1.

192         EventManager.getInstance().emit('stats:cleared');
                                       ~~~~~~~~~~~~~~~~~~~~~

  src/application/EventManager.ts:98:49
    98     emit<K extends keyof AppEventMap>(event: K, data: AppEventMap[K]): void {
                                                       ~~~~~~~~~~~~~~~~~~~~
    An argument for 'data' was not provided.

src/ui/formatting/FormatOptions.ts:710:5 - error TS2693: 'TextFormat' only refers to a type, but is being used as a value here.

710     TextFormat
        ~~~~~~~~~~

src/ui/modals/FilePickerModal.ts:324:13 - error TS2322: Type 'import("C:/Users/user/Documents/personal/.obsidian/plugins/SpeechNote/src/ui/components/FileValidator").ValidationResult' is not assignable to type 'import("C:/Users/user/Documents/personal/.obsidian/plugins/SpeechNote/src/ui/modals/FilePickerModal").ValidationResult'.
  Types of property 'errors' are incompatible.
    Type 'ValidationError[] | undefined' is not assignable to type 'string[] | undefined'.
      Type 'ValidationError[]' is not assignable to type 'string[]'.
        Type 'ValidationError' is not assignable to type 'string'.

324             return await this.validator.validate(file, buffer);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/modals/FilePickerModal.ts:326:58 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Error | undefined'.

326             this.logger?.error('File validation failed', error);
                                                             ~~~~~

src/ui/modals/FilePickerModal.ts:329:36 - error TS18046: 'error' is of type 'unknown'.

329                 errors: [`검증 실패: ${error.message}`]
                                       ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:108:47 - error TS2339: Property 'dispose' does not exist on type 'FileValidator | FileBrowser | DragDropZone | RecentFiles | ProgressIndicator | EventHandlers'.
  Property 'dispose' does not exist on type 'FileValidator'.

108             if (component && typeof component.dispose === 'function') {
                                                  ~~~~~~~

src/ui/modals/FilePickerModalRefactored.ts:258:56 - error TS2345: Argument of type '(e: KeyboardEvent) => void' is not assignable to parameter of type 'EventListener'.
  Types of parameters 'e' and 'evt' are incompatible.
    Type 'Event' is missing the following properties from type 'KeyboardEvent': altKey, charCode, code, ctrlKey, and 21 more.

258         this.eventManager.add(this.modalEl, 'keydown', handler);
                                                           ~~~~~~~

src/ui/modals/FilePickerModalRefactored.ts:296:57 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Error | undefined'.

296             this.logger?.error('File selection failed', error);
                                                            ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:297:37 - error TS18046: 'error' is of type 'unknown'.

297             new Notice(`파일 선택 실패: ${error.message}`);
                                        ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:350:13 - error TS2322: Type 'import("C:/Users/user/Documents/personal/.obsidian/plugins/SpeechNote/src/ui/components/FileValidator").ValidationResult' is not assignable to type 'import("C:/Users/user/Documents/personal/.obsidian/plugins/SpeechNote/src/ui/modals/FilePickerModalRefactored").ValidationResult'.
  Types of property 'errors' are incompatible.
    Type 'ValidationError[] | undefined' is not assignable to type 'string[] | undefined'.

350             return await this.components.validator.validate(file, buffer);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/modals/FilePickerModalRefactored.ts:352:58 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Error | undefined'.

352             this.logger?.error('File validation failed', error);
                                                             ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:355:36 - error TS18046: 'error' is of type 'unknown'.

355                 errors: [`검증 실패: ${error.message}`]
                                       ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:434:49 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'Error | undefined'.

434             this.logger?.error('Submit failed', error);
                                                    ~~~~~

src/ui/modals/FilePickerModalRefactored.ts:435:34 - error TS18046: 'error' is of type 'unknown'.

435             new Notice(`처리 실패: ${error.message}`);
                                     ~~~~~

src/ui/notifications/NotificationManager.ts:620:9 - error TS2739: Type '{}' is missing the following properties from type 'ProgressNotificationOptions': type, message

620         options: ProgressNotificationOptions = {}
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/notifications/NotificationManager.ts:1080:5 - error TS2416: Property 'on' in type 'NotificationManager' is not assignable to the same property in base type 'EventEmitter<DefaultEventMap>'.
  Type '{ (event: "show", listener: (notification: NotificationOptions) => void): Unsubscribe; (event: "dismiss", listener: (id: string) => void): Unsubscribe; (event: "action", listener: (id: string, action: string) => void): Unsubscribe; }' is not assignable to type '<K>(eventName: string | symbol, listener: (...args: any[]) => void) => this'.
    Type 'Unsubscribe' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'Unsubscribe'.

1080     on(event: 'show', listener: (notification: NotificationOptions) => void): Unsubscribe;
         ~~

src/ui/notifications/NotificationManager.ts:1081:5 - error TS2416: Property 'on' in type 'NotificationManager' is not assignable to the same property in base type 'EventEmitter<DefaultEventMap>'.
  Type '{ (event: "show", listener: (notification: NotificationOptions) => void): Unsubscribe; (event: "dismiss", listener: (id: string) => void): Unsubscribe; (event: "action", listener: (id: string, action: string) => void): Unsubscribe; }' is not assignable to type '<K>(eventName: string | symbol, listener: (...args: any[]) => void) => this'.

1081     on(event: 'dismiss', listener: (id: string) => void): Unsubscribe;
         ~~

src/ui/notifications/NotificationManager.ts:1082:5 - error TS2416: Property 'on' in type 'NotificationManager' is not assignable to the same property in base type 'EventEmitter<DefaultEventMap>'.
  Type '{ (event: "show", listener: (notification: NotificationOptions) => void): Unsubscribe; (event: "dismiss", listener: (id: string) => void): Unsubscribe; (event: "action", listener: (id: string, action: string) => void): Unsubscribe; }' is not assignable to type '<K>(eventName: string | symbol, listener: (...args: any[]) => void) => this'.

1082     on(event: 'action', listener: (id: string, action: string) => void): Unsubscribe;
         ~~

src/ui/notifications/NotificationManager.ts:1083:5 - error TS2416: Property 'on' in type 'NotificationManager' is not assignable to the same property in base type 'EventEmitter<DefaultEventMap>'.
  Type '{ (event: "show", listener: (notification: NotificationOptions) => void): Unsubscribe; (event: "dismiss", listener: (id: string) => void): Unsubscribe; (event: "action", listener: (id: string, action: string) => void): Unsubscribe; }' is not assignable to type '<K>(eventName: string | symbol, listener: (...args: any[]) => void) => this'.

1083     on(event: string, listener: (...args: any[]) => void): Unsubscribe {
         ~~

src/ui/progress/ProgressTracker.ts:183:14 - error TS2420: Class 'ProgressTracker' incorrectly implements interface 'IProgressTracker'.
  The types returned by 'on(...)' are incompatible between these types.
    Type 'this' is not assignable to type 'Unsubscribe'.
      Type 'ProgressTracker' is not assignable to type 'Unsubscribe'.
        Type 'ProgressTracker' provides no match for the signature '(): void'.

183 export class ProgressTracker extends EventEmitter implements IProgressTracker {
                 ~~~~~~~~~~~~~~~

src/ui/settings/base/CommonUIComponents.ts:62:17 - error TS2322: Type 'string | undefined' is not assignable to type 'string | number | boolean | null'.
  Type 'undefined' is not assignable to type 'string | number | boolean | null'.

62                 'aria-labelledby': label ? `progress-label-${Date.now()}` : undefined
                   ~~~~~~~~~~~~~~~~~

src/ui/settings/base/CommonUIComponents.ts:214:21 - error TS2322: Type 'string | undefined' is not assignable to type 'string | number | boolean | null'.

214                     'hidden': tab.id !== activeTab ? 'true' : undefined
                        ~~~~~~~~

src/ui/settings/base/CommonUIComponents.ts:323:41 - error TS7006: Parameter 'details' implicitly has an 'any' type.

323             errorEl.createEl('details', details => {
                                            ~~~~~~~

src/ui/settings/base/CommonUIComponents.ts:323:41 - error TS2345: Argument of type '(details: any) => void' is not assignable to parameter of type 'string | DomElementInfo | undefined'.

323             errorEl.createEl('details', details => {
                                            ~~~~~~~~~~~~

src/ui/settings/components/ApiKeyValidator.ts:77:17 - error TS18046: 'error' is of type 'unknown'.

77             if (error.status === 401) {
                   ~~~~~

src/ui/settings/components/ApiKeyValidator.ts:83:17 - error TS18046: 'error' is of type 'unknown'.

83             if (error.status === 429) {
                   ~~~~~

src/ui/settings/components/AudioSettings.ts:43:27 - error TS2345: Argument of type '(value: 'whisper-1') => Promise<void>' is not assignable to parameter of type '(value: string) => any'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'string' is not assignable to type '"whisper-1"'.

43                 .onChange(async (value: 'whisper-1') => {
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/components/GeneralSettings.ts:32:27 - error TS2345: Argument of type '(value: 'cursor' | 'end' | 'beginning') => Promise<void>' is not assignable to parameter of type '(value: string) => any'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'string' is not assignable to type '"cursor" | "end" | "beginning"'.

32                 .onChange(async (value: 'cursor' | 'end' | 'beginning') => {
                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/components/GeneralSettings.ts:76:31 - error TS2345: Argument of type '(value: 'none' | 'inline' | 'sidebar') => Promise<void>' is not assignable to parameter of type '(value: string) => any'.
  Types of parameters 'value' and 'value' are incompatible.
    Type 'string' is not assignable to type '"none" | "inline" | "sidebar"'.

76                     .onChange(async (value: 'none' | 'inline' | 'sidebar') => {
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/components/ProviderSettings.ts:619:46 - error TS2532: Object is possibly 'undefined'.

619         return this.apiKeys.has(provider) && this.apiKeys.get(provider)?.length > 0;
                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/components/ProviderSettings.ts:682:9 - error TS2322: Type 'string' is not assignable to type '"auto" | "whisper" | "deepgram" | undefined'.

682         this.plugin.settings.provider = provider;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/components/ProviderSettings.ts:696:9 - error TS2322: Type 'SelectionStrategy' is not assignable to type 'SelectionStrategy | undefined'.

696         this.plugin.settings.selectionStrategy = strategy;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/EnhancedSettingsTab.ts:31:30 - error TS2511: Cannot create an instance of an abstract class.

31         this.memoryManager = new AutoDisposable();
                                ~~~~~~~~~~~~~~~~~~~~

src/ui/settings/EnhancedSettingsTab.ts:49:28 - error TS2339: Property 'registerDisposable' does not exist on type 'AutoDisposable'.

49         this.memoryManager.registerDisposable({
                              ~~~~~~~~~~~~~~~~~~

src/ui/settings/error/ErrorHandling.ts:281:13 - error TS7053: Element implicitly has an 'any' type because expression of type '"type"' can't be used to index type '{ name: string; message: string; stack: string | undefined; timestamp: string; }'.
  Property 'type' does not exist on type '{ name: string; message: string; stack: string | undefined; timestamp: string; }'.

281             details['type'] = error.type;
                ~~~~~~~~~~~~~~~

src/ui/settings/error/ErrorHandling.ts:282:13 - error TS7053: Element implicitly has an 'any' type because expression of type '"recoverable"' can't be used to index type '{ name: string; message: string; stack: string | undefined; timestamp: string; }'.
  Property 'recoverable' does not exist on type '{ name: string; message: string; stack: string | undefined; timestamp: string; }'.

282             details['recoverable'] = error.recoverable;
                ~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/AdvancedSettingsPanel.ts:118:25 - error TS2322: Type 'string' is not assignable to type '"auto" | "none" | "manual" | undefined'.

118                         this.plugin.settings.fallbackStrategy = value;
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/AdvancedSettingsPanel.ts:385:29 - error TS2322: Type 'string' is not assignable to type '"all" | "latency" | "accuracy" | "cost" | undefined'.

385                             this.plugin.settings.abTestMetrics = value;
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/AdvancedSettingsPanel.ts:905:9 - error TS2322: Type 'SelectionStrategy.PERFORMANCE_OPTIMIZED' is not assignable to type 'SelectionStrategy | undefined'.

905         this.plugin.settings.selectionStrategy = SelectionStrategy.PERFORMANCE_OPTIMIZED;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/AdvancedSettingsPanel.ts:926:9 - error TS2322: Type 'SelectionStrategy' is not assignable to type 'SelectionStrategy | undefined'.

926         this.plugin.settings.selectionStrategy = strategy;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManager.ts:795:13 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

795             this.plugin.settings.apiKey = encrypted;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManager.ts:796:13 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

796             this.plugin.settings.whisperApiKey = encrypted;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManager.ts:798:13 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

798             this.plugin.settings.deepgramApiKey = encrypted;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManager.ts:838:64 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'EncryptedData'.

838                 const decrypted = await this.encryptor.decrypt(key);
                                                                   ~~~

src/ui/settings/provider/components/APIKeyManager.ts:851:64 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'EncryptedData'.

851                 const decrypted = await this.encryptor.decrypt(this.plugin.settings.deepgramApiKey);
                                                                   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:260:18 - error TS2339: Property 'debounce' does not exist on type 'APIKeyManagerRefactored'.

260             this.debounce(`validate-${provider}`, () => {
                     ~~~~~~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:701:17 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

701                 this.plugin.settings.apiKey = encrypted;
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:702:17 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

702                 this.plugin.settings.whisperApiKey = encrypted;
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:704:17 - error TS2322: Type 'EncryptedData' is not assignable to type 'string'.

704                 this.plugin.settings.deepgramApiKey = encrypted;
                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:897:68 - error TS2345: Argument of type 'string' is not assignable to parameter of type 'EncryptedData'.

897                     const decrypted = await this.encryptor.decrypt(key);
                                                                       ~~~

src/ui/settings/provider/components/APIKeyManagerRefactored.ts:941:22 - error TS2304: Cannot find name 'App'.

941     constructor(app: App, private plugin: SpeechToTextPlugin) {
                         ~~~

src/ui/settings/provider/ProviderSettingsContainer.ts:6:40 - error TS2307: Cannot find module './components/ProviderMetricsDisplay' or its corresponding type declarations.

6 import { ProviderMetricsDisplay } from './components/ProviderMetricsDisplay';
                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainer.ts:629:9 - error TS2322: Type 'SelectionStrategy.PERFORMANCE_OPTIMIZED' is not assignable to type 'SelectionStrategy | undefined'.

629         this.plugin.settings.selectionStrategy = SelectionStrategy.PERFORMANCE_OPTIMIZED;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainer.ts:660:9 - error TS2322: Type 'string' is not assignable to type '"auto" | "whisper" | "deepgram" | undefined'.

660         this.plugin.settings.provider = provider;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainer.ts:665:9 - error TS2322: Type 'SelectionStrategy' is not assignable to type 'SelectionStrategy | undefined'.

665         this.plugin.settings.selectionStrategy = strategy;
            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:398:13 - error TS2322: Type 'string' is not assignable to type '"auto" | "whisper" | "deepgram" | undefined'.

398             this.plugin.settings.provider = value;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:416:13 - error TS2322: Type 'SelectionStrategy' is not assignable to type 'SelectionStrategy | undefined'.

416             this.plugin.settings.selectionStrategy = strategy;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:648:13 - error TS2322: Type 'SelectionStrategy.PERFORMANCE_OPTIMIZED' is not assignable to type 'SelectionStrategy | undefined'.

648             this.plugin.settings.selectionStrategy = SelectionStrategy.PERFORMANCE_OPTIMIZED;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:720:28 - error TS2339: Property 'destroy' does not exist on type 'APIKeyManager'.

720         this.apiKeyManager.destroy?.();
                               ~~~~~~~

src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:721:28 - error TS2339: Property 'destroy' does not exist on type 'AdvancedSettingsPanel'.

721         this.advancedPanel.destroy?.();
                               ~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:20:14 - error TS2720: Class 'SettingsTabOptimized' incorrectly implements class 'AutoDisposable'. Did you mean to extend 'AutoDisposable' and inherit its members as a subclass?
  Property 'resourceManager' is missing in type 'SettingsTabOptimized' but required in type 'AutoDisposable'.

20 export class SettingsTabOptimized extends PluginSettingTab implements AutoDisposable {
                ~~~~~~~~~~~~~~~~~~~~

  src/utils/memory/MemoryManager.ts:473:15
    473     protected resourceManager = new ResourceManager();
                      ~~~~~~~~~~~~~~~
    'resourceManager' is declared here.

src/ui/settings/SettingsTabOptimized.ts:73:43 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'string | Error'.

73             this.errorManager.handleError(error, {
                                             ~~~~~

src/ui/settings/SettingsTabOptimized.ts:95:9 - error TS2322: Type '() => Promise<Promise<void>>' is not assignable to type '() => Promise<void>'.
  Type 'Promise<Promise<void>>' is not assignable to type 'Promise<void>'.
    Type 'Promise<void>' is not assignable to type 'void'.

95         this.saveSettings = debounceAsync(async () => {
           ~~~~~~~~~~~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:105:47 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'string | Error'.

105                 this.errorManager.handleError(error, {
                                                  ~~~~~

src/ui/settings/SettingsTabOptimized.ts:345:13 - error TS2564: Property 'saveSettings' has no initializer and is not definitely assigned in the constructor.

345     private saveSettings: () => Promise<void>;
                ~~~~~~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:535:13 - error TS2564: Property 'inputEl' has no initializer and is not definitely assigned in the constructor.

535     private inputEl: HTMLInputElement;
                ~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:536:13 - error TS2564: Property 'toggleBtn' has no initializer and is not definitely assigned in the constructor.

536     private toggleBtn: HTMLButtonElement;
                ~~~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:537:13 - error TS2564: Property 'validateBtn' has no initializer and is not definitely assigned in the constructor.

537     private validateBtn: HTMLButtonElement;
                ~~~~~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:714:29 - error TS2551: Property 'settingsTab' does not exist on type 'SpeechToTextPlugin'. Did you mean 'settings'?

714                 this.plugin.settingsTab.display();
                                ~~~~~~~~~~~

  src/main.ts:18:5
    18     settings!: SpeechToTextSettings;
           ~~~~~~~~
    'settings' is declared here.

src/ui/settings/SettingsTabOptimized.ts:730:48 - error TS2339: Property 'getDefaultSettings' does not exist on type 'SpeechToTextPlugin'.

730             this.plugin.settings = this.plugin.getDefaultSettings();
                                                   ~~~~~~~~~~~~~~~~~~

src/ui/settings/SettingsTabOptimized.ts:736:25 - error TS2551: Property 'settingsTab' does not exist on type 'SpeechToTextPlugin'. Did you mean 'settings'?

736             this.plugin.settingsTab.display();
                            ~~~~~~~~~~~

  src/main.ts:18:5
    18     settings!: SpeechToTextSettings;
           ~~~~~~~~
    'settings' is declared here.

src/ui/settings/types/SettingsTypes.ts:373:63 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'T[keyof T]'.

373             const result = this.validateField(key as keyof T, value);
                                                                  ~~~~~

src/ui/settings/types/SettingsTypes.ts:408:17 - error TS2322: Type '{}' is not assignable to type 'T[Extract<keyof T, string>]'.

408                 result[key] = deepMerge(target[key] || {}, source[key] as any);
                    ~~~~~~~~~~~

src/utils/async/AsyncTaskCoordinator.ts:278:17 - error TS2741: Property '[Symbol.toStringTag]' is missing in type 'CancellablePromise<T>' but required in type 'Promise<T>'.

278             let promise: Promise<T> = this.cancellablePromise;
                    ~~~~~~~

  node_modules/typescript/lib/lib.es2015.symbol.wellknown.d.ts:174:14
    174     readonly [Symbol.toStringTag]: string;
                     ~~~~~~~~~~~~~~~~~~~~
    '[Symbol.toStringTag]' is declared here.

src/utils/common/helpers.ts:169:40 - error TS2345: Argument of type 'T[Extract<keyof T, string>] | undefined' is not assignable to parameter of type 'Partial<T[Extract<keyof T, string>]>'.
  Type 'undefined' is not assignable to type 'Partial<T[Extract<keyof T, string>]>'.

169                 deepMerge(target[key], source[key]);
                                           ~~~~~~~~~~~

src/utils/memory/MemoryManager.ts:207:36 - error TS2344: Type 'string' does not satisfy the constraint 'object'.

207     private elements = new WeakMap<string, WeakRef<HTMLElement>>();
                                       ~~~~~~


Found 80 errors in 25 files.

Errors  Files
     2  src/application/EventManager.ts:311
     2  src/infrastructure/api/SettingsAPI.ts:285
     1  src/ui/dashboard/StatisticsDashboard.ts:192
     1  src/ui/formatting/FormatOptions.ts:710
     3  src/ui/modals/FilePickerModal.ts:324
     9  src/ui/modals/FilePickerModalRefactored.ts:108
     5  src/ui/notifications/NotificationManager.ts:620
     1  src/ui/progress/ProgressTracker.ts:183
     4  src/ui/settings/base/CommonUIComponents.ts:62
     2  src/ui/settings/components/ApiKeyValidator.ts:77
     1  src/ui/settings/components/AudioSettings.ts:43
     2  src/ui/settings/components/GeneralSettings.ts:32
     3  src/ui/settings/components/ProviderSettings.ts:619
     2  src/ui/settings/EnhancedSettingsTab.ts:31
     2  src/ui/settings/error/ErrorHandling.ts:281
     4  src/ui/settings/provider/components/AdvancedSettingsPanel.ts:118
     5  src/ui/settings/provider/components/APIKeyManager.ts:795
     6  src/ui/settings/provider/components/APIKeyManagerRefactored.ts:260
     4  src/ui/settings/provider/ProviderSettingsContainer.ts:6
     5  src/ui/settings/provider/ProviderSettingsContainerRefactored.ts:398
    11  src/ui/settings/SettingsTabOptimized.ts:20
     2  src/ui/settings/types/SettingsTypes.ts:373
     1  src/utils/async/AsyncTaskCoordinator.ts:278
     1  src/utils/common/helpers.ts:169
     1  src/utils/memory/MemoryManager.ts:207
PS C:\Users\user\Documents\personal\.obsidian\plugins\SpeechNote>

--- a/package.json
+++ b/package.json
@@ -5,7 +5,7 @@
   "main": "main.js",
   "scripts": {
     "dev": "node esbuild.config.mjs",
-    "build": "tsc -noEmit -skipLibCheck && node esbuild.config.mjs production",
+    "build": "npx tsc -noEmit -skipLibCheck && node esbuild.config.mjs production",
     "lint": "eslint src/**/*.ts",
     "lint:fix": "eslint src/**/*.ts --fix",
     "format": "prettier --write 'src/**/*.{ts,tsx}'",
@@ -29,7 +29,7 @@
     "version": "node version-bump.mjs && git add manifest.json versions.json",
     "clean": "rm -rf main.js main.js.map coverage .jest-cache",
     "clean:all": "npm run clean && rm -rf node_modules",
-    "typecheck": "tsc -noEmit",
+    "typecheck": "npx tsc -noEmit",
     "validate": "npm run lint && npm run typecheck && npm run test",
     "ci": "npm run validate && npm run build"
   },

